### PR TITLE
Enhancement: Add support for GitHub actions

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds support for [GitHub actions](https://github.com/features/actions)

💁‍♂ For reference, see [Virtual environments for GitHub Actions: Environment Variables](https://help.github.com/en/articles/virtual-environments-for-github-actions#environment-variables).

### CODE REVIEW

**Comment by localheinz** [comment]
Build failure appears to be unrelated.

File: codecov [hunk 1]
```diff
@@ -735,6 +735,18 @@ then
...
+then
+  say "$e==>$x GitHub Actions detected."
+
+  # https://github.com/features/actions
+  service="github-actions"
+
+  # https://help.github.com/en/articles/virtual-environments-for-github-actions#environment-variables
+  branch="${GITHUB_REF#/refs/heads/}"
+  commit="${GITHUB_SHA}"
+  slug="${GITHUB_REPOSITORY}"
```
Comment (localheinz) [inline codecov]: Not sure, does this information suffice?

Or do we need to specify the `CODECOV_TOKEN` via environment variable or `-t` option?

**Comment by Wikunia** [comment]
What needs to be done to get this merged? 
Relevant i.e. for https://github.com/julia-actions/julia-uploadcodecov/issues/1

**Comment by drazisil** [comment]
@localheinz Thank you for this PR. Your tests pass, so I'm going to merge this.

**Comment by localheinz** [comment]
Thank yoy, @drazisil!

**Comment by drazisil** [comment]
@localheinz 

As a heads up, I did not realize that we needed to add this service tag to the upload backend as well. So this might throw errors until I get that PR merged. Will hopefully be soon.

**Pull request merged**